src/SensitiveFloat.jl

```diff
@@ -0,0 +1,275 @@
+
+export multiply_sfs!, add_scaled_sfs!, combine_sfs!, add_sources_sf!
+
+@doc """
+A function value and its derivative with respect to its arguments.
+
+Attributes:
+  v:  The value
+  d:  The derivative with respect to each variable in
+      P-dimensional VariationalParams for each of S celestial objects
+      in a local_P x local_S matrix.
+  h:  The second derivative with respect to each variational parameter,
+      in the same format as d.  This is used for the full Hessian
+      with respect to all the sources.
+  hs: An array of per-source Hessians.  This will generally be reserved
+      for the Hessian of brightness values that depend only on one source.
+""" ->
+type SensitiveFloat{ParamType <: CelesteTypes.ParamSet, NumType <: Number}
+    # Actually a single value, but an Array to avoid memory allocation
+    v::Vector{NumType}
```

Comment:
I didn't try.  Do you think it's important?

On Tue, Mar 8, 2016 at 12:04 PM, Jeffrey Regier <notifications@github.com>
wrote:

> In src/SensitiveFloat.jl
> <https://github.com/jeff-regier/Celeste.jl/pull/137#discussion_r55418802>:
>
> > +A function value and its derivative with respect to its arguments.
> > +
> > +Attributes:
> > +  v:  The value
> > +  d:  The derivative with respect to each variable in
> > +      P-dimensional VariationalParams for each of S celestial objects
> > +      in a local_P x local_S matrix.
> > +  h:  The second derivative with respect to each variational parameter,
> > +      in the same format as d.  This is used for the full Hessian
> > +      with respect to all the sources.
> > +  hs: An array of per-source Hessians.  This will generally be reserved
> > +      for the Hessian of brightness values that depend only on one source.
> > +""" ->
> > +type SensitiveFloat{ParamType <: CelesteTypes.ParamSet, NumType <: Number}
> > +    # Actually a single value, but an Array to avoid memory allocation
> > +    v::Vector{NumType}
>
> Any luck duplicating the phenomenon---memory allocation on access---in a
> minimal working example?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jeff-regier/Celeste.jl/pull/137/files#r55418802>.
>